src/fsharp/fsi/fsi.fs

```diff
@@ -1161,30 +1162,21 @@ type internal FsiDynamicCompiler
                         let symbol = FSharpSymbol.Create(cenv, v.Item)
                         let symbolUse = FSharpSymbolUse(tcGlobals, newState.tcState.TcEnvFromImpls.DisplayEnv, symbol, ItemOccurence.Binding, v.DeclarationLocation)
                         fsi.TriggerEvaluation (fsiValueOpt, symbolUse, decl)
+                        lastValue <- fsiValueOpt
                     | FSharpImplementationFileDeclaration.Entity (e,_) ->
                         // Report a top-level module or namespace definition
                         let symbol = FSharpSymbol.Create(cenv, e.Item)
                         let symbolUse = FSharpSymbolUse(tcGlobals, newState.tcState.TcEnvFromImpls.DisplayEnv, symbol, ItemOccurence.Binding, e.DeclarationLocation)
                         fsi.TriggerEvaluation (None, symbolUse, decl)
+                        lastValue <- None
                     | FSharpImplementationFileDeclaration.InitAction _ ->
                         // Top level 'do' bindings are not reported as incremental declarations
```

Comment:
Probably need a lastValue <- None here too.  Unless the InitAction is guaranteed to never follow MemberOrFunctionOrValue  or FSharpImplementationFileDeclaration

Reply:
I had a similar thought; the only reason I didn't set `None` there is because I was following the event `fsi.TriggerEvaluation` and that event isn't fired in `InitAction`.

I'll set `lastValue <- None` in `InitAction` for completeness.